test/nix/initramfs.nix

```diff
@@ -26,6 +30,24 @@ in stdenvNoCC.mkDerivation {
 
     cp -r ${etc}/* $out/etc/
 
+    RESOLV_CONF_FILE="$out/etc/resolv.conf"
+    if [ -n "${asterDns}" ] && [ "${asterDns}" != "none" ]; then
+      echo "nameserver ${asterDns}" > "$RESOLV_CONF_FILE"
+    else
+      if [ ! -f "${host_resolv_conf}" ]; then
+        echo "/etc/resolv.conf does not exist. resolv.conf is not generated"
+      else
+        FIRST_NAMESERVER=$(grep -oP '(?<=^nameserver\s)\S+' ${host_resolv_conf} | head -n 1)
+
+        if [ "$FIRST_NAMESERVER" = "127.0.0.53" ]; then
+          echo "Host DNS is managed by man:systemd-resolved(8). resolv.conf is not generated"
```

Comment:
What is the motivation of checking this specific DNS server? Do you expect users to give this DNS (by mistake)?